src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -0,0 +1,237 @@
+    return new byte[]{spawnCount, landDensity, plateauDensity, mountainDensity, rampDensity};
+  }
+
+  public void onCloseButtonClicked() {
+    onCloseButtonClickedListener.run();
+  }
+
+  public void onGenerateMapButtonClicked() {
+    onGenerateMap();
+  }
+
+  public void onGenerateMap() {
+      if (!previousMapName.getText().isEmpty()) {
+        mapGeneratorService.generateMap(previousMapName.getText())
+            .exceptionally(throwable -> {
+              handleGenerationException(throwable);
+              return null;
+            })
+            .thenAccept(mapName -> Platform.runLater(() -> {
+              createGameController.initMapSelection();
+              createGameController.mapListView.getItems().stream()
+                  .filter(mapBean -> mapBean.getFolderName().equalsIgnoreCase(mapName))
+                  .findAny().ifPresent(mapBean -> {
+                createGameController.mapListView.getSelectionModel().select(mapBean);
+                createGameController.mapListView.scrollTo(mapBean);
+                createGameController.setSelectedMap(mapBean);
+              });
+            }));
+      } else {
+        byte[] optionArray = getOptionArray();
+        mapGeneratorService.generateMap(optionArray)
+            .exceptionally(throwable -> {
+              handleGenerationException(throwable);
+              return null;
+            })
+            .thenAccept(mapName -> Platform.runLater(() -> {
+              createGameController.initMapSelection();
+              createGameController.mapListView.getItems().stream()
+                  .filter(mapBean -> mapBean.getFolderName().equalsIgnoreCase(mapName))
+                  .findAny().ifPresent(mapBean -> {
+                createGameController.mapListView.getSelectionModel().select(mapBean);
+                createGameController.mapListView.scrollTo(mapBean);
+                createGameController.setSelectedMap(mapBean);
+              });
+            }));
+      }
+    onCloseButtonClickedListener.run();
+  }
+
+  private void handleGenerationException(Throwable e) {
+    Throwable cause = e.getCause();
+    if (cause instanceof IllegalArgumentException) {
+      String msg = cause.getMessage();
+      if (msg.startsWith("New Version")) {
+        notificationService.addImmediateErrorNotification(cause, "mapGenerator.tooNewVersion");
+        log.error("Map generation failed", cause);
+      } else if (msg.startsWith("Old Version")) {
+        notificationService.addImmediateErrorNotification(cause, "mapGenerator.tooOldVersion");
+        log.error("Map generation failed", cause);
+      } else {
+        notificationService.addImmediateErrorNotification(e, "mapGenerator.generationFailed");
+        log.error("Map generation failed", e);
+      }
+    } else {
+      notificationService.addImmediateErrorNotification(e, "mapGenerator.generationFailed");
+      log.error("Map generation failed", e);
```

Comment:
Should they not all be warn

Reply:
I am not sure what you me by all of them being warn? is there another function to the notification service?

Reply:
log.warn(....)

Reply:
Ah got it

Reply:
Done